warehouse/mssql/mssql.go

```diff
@@ -0,0 +1,662 @@
+			return nil, err
+		}
+		ObjectPath := tmpDirPath + dirName + fmt.Sprintf(`%s_%s_%d/`, ms.Warehouse.Destination.DestinationDefinition.Name, ms.Warehouse.Destination.ID, time.Now().Unix()) + object
+		err = os.MkdirAll(filepath.Dir(ObjectPath), os.ModePerm)
+		if err != nil {
+			pkgLogger.Errorf("MS: Error in making tmp directory for downloading load file for table:%s: %s, %s %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		objectFile, err := os.Create(ObjectPath)
+		if err != nil {
+			pkgLogger.Errorf("MS: Error in creating file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		downloader, err := filemanager.New(&filemanager.SettingsT{
+			Provider: warehouseutils.ObjectStorageType(ms.Warehouse.Destination.DestinationDefinition.Name, ms.Warehouse.Destination.Config),
+			Config:   ms.Warehouse.Destination.Config,
+		})
+		if err != nil {
+			pkgLogger.Errorf("MS: Error in setting up a downloader for destionationID : %s Error : %v", ms.Warehouse.Destination.ID, err)
+			return nil, err
+		}
+		err = downloader.Download(objectFile, object)
+		if err != nil {
+			pkgLogger.Errorf("MS: Error in downloading file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		fileName := objectFile.Name()
+		if err = objectFile.Close(); err != nil {
+			pkgLogger.Errorf("MS: Error in closing downloaded file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		fileNames = append(fileNames, fileName)
+	}
+	return fileNames, nil
+
+}
+
+func (ms *HandleT) loadTable(tableName string, tableSchemaInUpload warehouseutils.TableSchemaT, skipTempTableDelete bool) (stagingTableName string, err error) {
+	pkgLogger.Infof("MS: Starting load for table:%s", tableName)
+
+	// sort column names
+	sortedColumnKeys := warehouseutils.SortColumnKeysFromColumnMap(tableSchemaInUpload)
+	sortedColumnString := strings.Join(sortedColumnKeys, ", ")
+
+	fileNames, err := ms.DownloadLoadFiles(tableName)
+	defer misc.RemoveFilePaths(fileNames...)
+	if err != nil {
+		return
+	}
+
+	txn, err := ms.Db.Begin()
+	if err != nil {
+		pkgLogger.Errorf("MS: Error while beginning a transaction in db for loading in table:%s: %v", tableName, err)
+		return
+	}
+	// create temporary table
+	stagingTableName = fmt.Sprintf(`%s%s_%s`, stagingTablePrefix, tableName, strings.Replace(uuid.NewV4().String(), "-", "", -1))
+	//sqlStatement := fmt.Sprintf(`CREATE TEMPORARY TABLE "%[2]s" (LIKE "%[1]s"."%[3]s")`, ms.Namespace, stagingTableName, tableName)
+	//sqlStatement := fmt.Sprintf(`CREATE  TABLE ##%[2]s like %[1]s.%[3]s`, ms.Namespace, stagingTableName, tableName)
+	sqlStatement := fmt.Sprintf(`select top 0 * into %[1]s.%[2]s from %[1]s.%[3]s`, ms.Namespace, stagingTableName, tableName)
```

Comment:
creating temp table does not work in this case? even with permissions granted? https://stackoverflow.com/a/37440178/3921672

Reply:
creation was fine. But since the loadTable creates a stagingTableName table and commits the txn. Was not able to access the same table passed to this method: loadUserTables. Hence resorted to regular tables.

Reply:
@bipin-rudder does the temp table not get deleted when connection is closed. The `loadTable` can also return the temp table name and the connection handle. Check snowflake or redshift implementation

Reply:
Using individual db.exec or txn.exec(After the txn.commit, table is purged) is not working. 
SQL prepared stmts does not seem to support temp objects : https://github.com/denisenkom/go-mssqldb/issues/149
Commented the same in code.